### PR TITLE
Upgrade Erlang to R22

### PR DESCRIPTION
**Pull request by dunielpls**
This PR upgrades erlang and friends to run on OTP R22 (latest is 22.0.2 at the time of writing).

### CODE REVIEW

**Comment by dunielpls** [comment]
There was a bit of a mess the JInterface application. I had bumped the OpenJDK dependency to 11, but in the heat of the moment I forgot that it is unavailable for x86, and the CI got mad at me.

So tl;dr: the force push rectifies my mistake.

**Comment by dunielpls** [comment]
There was a bit of a mess the JInterface application. I had bumped the OpenJDK dependency to 11, but in the heat of the moment I forgot that it is unavailable for x86, and the CI got mad at me.

So tl;dr: the force push rectifies my mistake.

**Comment by dunielpls** [comment]
There was a bit of a mess the JInterface application. I had bumped the OpenJDK dependency to 11, but in the heat of the moment I forgot that it is unavailable for x86, and the CI got mad at me.

So tl;dr: the force pushes rectify my mistake.

**Comment by dunielpls** [comment]
There was a bit of a mess the JInterface application. I had bumped the OpenJDK dependency to 11, but in the heat of the moment I forgot that it is unavailable for x86, and the CI got mad at me.

So tl;dr: the force pushes rectify my mistake.

File: community/erlang/APKBUILD [hunk 1]
```diff
@@ -1,105 +1,99 @@
...
-	     $pkgname-snmp:snmp
-	     $pkgname-ssh:ssh
-	     $pkgname-ssl:ssl
-	     $pkgname-stdlib:stdlib
-	     $pkgname-syntax-tools:syntax_tools
-	     $pkgname-tools:tools
-	     $pkgname-xmerl:xmerl
-	     $pkgname-wx:wx"
-options="!check"
+             $pkgname-common-test:common_test
```
Comment (Ikke) [inline community/erlang/APKBUILD]: aports uses tabs for indentation, not spaces.

File: community/erlang/APKBUILD [hunk 2]
```diff
@@ -138,47 +131,47 @@ ssh() { _mv_erlang_lib ssh; depends="$depends erlang-public-key erlang-inets"; }
...
+            mkdir -p "$d"
+            mv "$pkgdir/$i" "$d"
+            rmdir "$pkgdir/${i%/*}" 2>/dev/null || true
+        fi
+    done
+    # move *.so links needed when linking the apps to -dev packages
+    for i in lib/*.so usr/lib/*.so; do
+        if [ -L "$i" ]; then
+            mkdir -p "$subpkgdir"/"${i%/*}"
+            mv "$i" "$subpkgdir/$i" || return 1
```
Comment (Ikke) [inline community/erlang/APKBUILD]: If you're improving the APKBUILD anyway, these `|| return 1` can be removed.

File: community/erlang/APKBUILD (hunk 1 above)
Comment (dunielpls) [reply community/erlang/APKBUILD]: Ah crap, newly configured editor. I'll fix that

File: community/erlang/APKBUILD (hunk 2 above)
Comment (dunielpls) [reply community/erlang/APKBUILD]: Point was primarily to get it going for now, but I'll remove it while I'm at it.

**Comment by dunielpls** [comment]
The `elixir` package has successfully been rebuilt locally by myself (x86_64) and mps on IRC (armv7) - I suspect the Drone build fails because of some peculiarities in the build environment.

**Comment by dunielpls** [comment]
Opened an issue upstream: https://github.com/elixir-lang/elixir/issues/9137

**Comment by dunielpls** [comment]
Rebased on top of commit 3697203876f8fcc803860efacc97b0fbd6cc0227 by @ncopa which disabled `testing/rebar3` on `armv7`.

Ready for merge once CI completes. Both should pass as a bug in the tests were fixed upstream in elixir-lang/elixir.

**Pull request closed without merging**